content/shell/android/shell_apk/src/org/chromium/content_shell_apk/ContentShellActivity.java

```diff
@@ -73,10 +75,7 @@ protected void onCreate(final Bundle savedInstanceState) {
         mWindowAndroid.setAnimationPlaceholderView(
                 mShellManager.getContentViewRenderView().getSurfaceView());
 
-        // TODO(cobalt, b/376148547): set Chrobalt initial url and remove this function.
-        if (mStartupUrl.isEmpty()) {
-            mStartupUrl = getUrlFromIntent(getIntent());
-        }
+        mStartupUrl = getUrlFromIntent(getIntent());
```

Comment:
@johnxwork rather than modifying contentshell original, can we just _copy_ this activity code into our directory and modify it there ? It's 237 lines, and i will expect we'll run into more customization issues soon. Rather than trying to refactor and exend it, copying seems like sensible and easy option.

Reply:
This will be addressed by https://github.com/youtube/cobalt/pull/4336.